### PR TITLE
Fix tests w/ pytest 3 and add HTTP proxy example

### PR DESCRIPTION
**Pull request by henryprecheur**
I fixed the monocle tests to run correctly with pytest 3, and I added a function in the examples on how to make an HTTP request using a proxy with Twisted's HTTP client.

https://saucedev.atlassian.net/browse/SC-794

No functionality change, I just needed to figure out how to do it =)

@sah 
@saucelabs/connected-cloud

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +135,30 @@ helpful to think of `yield` as in *traffic*.  `yield conn.read(10)` in
 an o-routine means "yield to other o-routines until we finish reading
 10 bytes".
 
+## Developer information
+
+To run individuals tests on your computer install py.test and the packages for
```
Comment (cstyan) [inline README.md]: `indivdual`

File: README.md [hunk 2]
```diff
@@ -135,6 +135,30 @@ helpful to think of `yield` as in *traffic*.  `yield conn.read(10)` in
...
+    $ cd monocle/
+    $ virtualenv --clear --no-site-package .venv
+    $ .venv/bin/pip install pytest twisted
+    ...
+
+You run the tests like this:
+
+    $ .venv/bin/python o_test.py twisted tests/
+
+Alternatively you can you tox to run all the tests for the different backends:
```
Comment (cstyan) [inline README.md]: `you can use`

Comment (henryprecheur) [inline README.md]: me no speak english very well

File: README.md (hunk 1 above)
Comment (henryprecheur) [inline README.md]: individual... ;-)

File: o_test.py [hunk 3]
```diff
@@ -217,10 +217,4 @@ def isinitpath(self, fn):
...
 
 if __name__ == '__main__':
-    try:
-        main(sys.argv[1:])
-    except:
-        traceback.print_exc(file=sys.stdout)
-        raise
-    finally:
-        sys.exit(_fail_count)
+    main(sys.argv[1:])
```
Comment (sah) [inline o_test.py]: I don't understand why you've removed the code to catch and print exceptions, and to set the exit code in the way test runners tend to do.

Comment (henryprecheur) [inline o_test.py]: I'll put it back

**Pull request merged**